options.go

```diff
@@ -0,0 +1,79 @@
+package yaml
+
+import (
+	"errors"
+	"io"
+)
+
+// config holds configuration options for YAML processing.
+//
+// It allows customization of various aspects of YAML parsing and serialization.
+type config struct {
+	// indent controls the number of spaces to use for indentation
+	indent *int
+	// knownFields enables strict field checking during unmarshalling
+	knownFields *bool
+
+	writer *io.Writer
+
+	// compactSequenceIndent determines if sequence indicators are part of indentation
+	compactSequenceIndent *bool
+}
+
+func newConfig(opts ...Option) (*config, error) {
+	c := &config{}
+	for _, opt := range opts {
+		if err := opt(c); err != nil {
+			return nil, err
+		}
+	}
+	return c, nil
+}
+
+type Option func(*config) error
```

Comment:
@ccoVeille could we separate Encoder and Decoder options?

Reply:
I don't think it is a good idea, because we can have an option that can apply for both, like FallbackToJSON.

Reply:
Someone want to add things to go-yaml CLI, haha !

I'm fine with the idea.

Having sub structs is indeed an option.

So we can pass only the minimal struct to encode, decode, node.decode, Marshall, Unmarshal.

Reply:
IMO the slight complexity added by interfaces is well worth it to disallow using options incorrectly at compare time. On the flip side, the methods can operate directly on the Encoder/Decoder so we don't need a separate config. For example: https://gist.github.com/RaduBerinde/5af78b7632997d251edcc50996070e24

Reply:
You are right.

It's close to what @inteon suggested here by taking a look at what was done in kubernetes options

- https://github.com/yaml/go-yaml/pull/162#discussion_r2481719207

Maybe not, at least the gist you worked is clear and simple.

I will try to work on this.

I feel like this should be addressed before merging this PR

Reply:
I have worked on providing a simple implementation, something simpler than what you suggested @RaduBerinde 

I decided to provide something simpler.